### PR TITLE
Refactoring Mindfulness Page [1/n]

### PR DESCRIPTION
**Pull request by Shaomei**
Refactor the mindfulness page to reduce code redundancy and separate data from the code. This is the first of some upcoming changes to this page as I am planning to componentize elements and the entire page later.

List of changes:
1. Create data/ folder to store static data used in the app in JSON formate, and separate data from the code. It now  contains only one JSON file (videos.json) that stores the meta data for all the mindfulness videos. We will later create separate json files for other static data such as default affirmation messages and text for handwave buttons. 
2. Use `array.map` to loop over available videos and reduce code redundancy;
3. Update video label so they are more readable for screen readers as aria-labels;
4. Add `key` attribute to video player component as recommended by React warning messages (see below). I followed the instructions [here](https://react.dev/learn/rendering-lists#keeping-list-items-in-order-with-key). 
Screenshot 2024-04-16 at 7 01 39 PM

5. Update and simplify related unit tests (see `__tests__/mindfulness.test.tsx`) using `array.map` as well.

### CODE REVIEW

**Comment by Shaomei** [comment]
All tests passed locally, see logs below:

```
~/g/ZoomCompanion on refactor:main ⇡1 *1 ❯ yarn test                                            Py base at 20:45:29
yarn run v1.22.11
$ jest
(node:74430) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
 PASS  __tests__/mindfulness.test.tsx
 PASS  __tests__/nametag.test.tsx
 PASS  __tests__/login.test.tsx
 PASS  __tests__/affirmation.test.tsx

Test Suites: 4 passed, 4 total
Tests:       10 passed, 10 total
Snapshots:   0 total
Time:        1.389 s
Ran all test suites.
✨  Done in 3.15s.

```

**Comment by Shaomei** [comment]
Amended some smalls changes to fix the build error/warnings. `npm run build` passed successfully now.

```
~/g/ZoomCompanion on refactor:main ⇡1 *1 !2 ❯ npm run build   took 6s Py base at 21:08:11

> next-test-zoom-app@0.1.0 build
> next build

   ▲ Next.js 14.0.4
   - Environments: .env.local

 ✓ Creating an optimized production build    
 ✓ Compiled successfully

./app/main/NameTag.tsx
77:6  Warning: React Hook useEffect has a missing dependency: 'debouncedEffect'. Either include it or remove the dependency array.  react-hooks/exhaustive-deps

info  - Need to disable some ESLint rules? Learn more here: https://nextjs.org/docs/basic-features/eslint#disabling-rules
 ✓ Linting and checking validity of types    
   Collecting page data  ...(node:75034) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
 ✓ Collecting page data    
(node:75033) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
   Generating static pages (0/11)  [    ]https://zoom.us/oauth/authorize?response_type=code&client_id=yptfVYBcQw_pDb_39jA_Q&redirect_uri=https%3A%2F%2Fhumpback-immune-frankly.ngrok-free.app%2Fpost-installation&code_challenge=YJMNbwArWCRIHfQZnJF45LGd0n0XMi6aLCYRje-Wy14&code_challenge_method=S256&state=WZV8VDqwmDedZL49KckRQ3fgvlK91a4hCqh8Cn0DXBk%3D
 ✓ Generating static pages (11/11) 
(node:75037) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
 ✓ Collecting build traces    
 ✓ Finalizing page optimization    

Route (app)                              Size     First Load JS
┌ ○ /                                    11.6 kB        93.5 kB
├ ○ /_not-found                          875 B          82.8 kB
├ λ /api/auth/[...nextauth]              0 B                0 B
├ λ /api/auth/users                      0 B                0 B
├ ○ /api/install                         0 B                0 B
├ ○ /install                             774 B          88.2 kB
├ ○ /main                                54.1 kB         136 kB
├ ○ /post-installation                   760 B          88.2 kB
└ ○ /sign-up                             1.38 kB        83.3 kB
+ First Load JS shared by all            82 kB
  ├ chunks/938-1e6442e2fe203f0c.js       26.7 kB
  ├ chunks/fd9d1056-9f67c486d101ba3a.js  53.3 kB
  ├ chunks/main-app-5ce6103bc2c0360b.js  222 B
  └ chunks/webpack-2169da390f4c8771.js   1.77 kB


○  (Static)   prerendered as static content
λ  (Dynamic)  server-rendered on demand using Node.js


```

File: lib/data.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+// Static data used in the app
+
+export const videoList = [{
+	name: 'guided breathing',
+	youtubeId: 'AKN_gbj8UtU',
+	description: 'One minute guided breathing to calm your breath',
+	alt: 'One minute guided breathing meditation video, with both audio guide and on-screen text guide',
+	label: 'guided-breathing-video'
```
Comment (blickly) [inline lib/data.ts:8]: What does the label do?

File: lib/data.ts
Comment (blickly) [inline lib/data.ts:1]: I'm not sure this is really interesting enough to put in its own file, and it definitely doesn't seem like a reusable "library" file.  Maybe just leave this array inside `Mindfulness.tsx` for now?

File: app/main/Mindfulness.tsx [hunk 2]
```diff
@@ -3,39 +3,32 @@
 "use client";
 
 import '../css/Mindfulness.css'; // Import CSS file
```
Comment (blickly) [inline app/main/Mindfulness.tsx:5]: I don't think these comments pull their weight. I'd remove them.

File: app/main/Mindfulness.tsx [hunk 3]
```diff
@@ -3,39 +3,32 @@
...
-  const youtube2ID = 'c7IkEhKtUwI'
-  const youtube3ID = 'sG69_omRCyo'
-  const youtube1Label = 'youtube-video-1';
-  const youtube2Label = 'youtube-video-2';
-  const youtube3Label = 'youtube-video-3';
-
   return (
     <div className="mindfulness-container">
       <h2 className="title">Mindfulness</h2>
       <div className="video-container">
```
Comment (blickly) [inline app/main/Mindfulness.tsx:28]: I was thinking that this "Video Container" thing might be a component, where the `videoList` can be passed as its argument.

File: app/main/Mindfulness.tsx [hunk 4]
```diff
@@ -3,39 +3,32 @@
...
       className='video'
       title='Youtube player'
-      aria-label={Label}
+      aria-label={label}
       src={`https://www.youtube.com/embed/${videoId}`}
     ></iframe>
   </div>
 );
 
 function Mindfullness() {
```
Comment (blickly) [inline app/main/Mindfulness.tsx:24]: Would it be clearer to name this something like `MindfullnessTabContents`?

File: app/main/Mindfulness.tsx [hunk 5]
```diff
@@ -3,39 +3,32 @@
...
+  label: string;
 }
 
-const generateVideoPlayer = ({ videoId, Label }: VideoPlayerProps) => (
-  <div className="video-wrapper">
+const generateVideoPlayer = ({ videoId, label }: VideoPlayerProps) => (
+  <div className="video-wrapper" key={label}>
     <iframe
       className='video'
       title='Youtube player'
```
Comment (blickly) [inline app/main/Mindfulness.tsx:17]: This title seems useless. Remove or update?

File: app/main/Mindfulness.tsx
Comment (blickly) [inline app/main/Mindfulness.tsx:1]: After this, I think we can count this file as reviewed and remove the line 1 comment.

**Review by blickly — CHANGES_REQUESTED** [review]

File: app/main/Mindfulness.tsx (hunk 4 above)
Comment (Shaomei) [reply app/main/Mindfulness.tsx:28]: I updated the aria-label to be alt-text for the video.

File: app/main/Mindfulness.tsx (hunk 3 above)
Comment (Shaomei) [reply app/main/Mindfulness.tsx:32]: Yeah that's a good idea for the next PR with the components.

File: lib/data.ts
Comment (Shaomei) [reply lib/data.ts:1]: Moved to a .json file.

File: app/main/Mindfulness.tsx (hunk 5 above)
Comment (Shaomei) [reply app/main/Mindfulness.tsx]: Updated it to video name.

File: app/main/Mindfulness.tsx
Comment (Shaomei) [reply app/main/Mindfulness.tsx:1]: SGTM

**Review by blickly — APPROVED** [review]

**Pull request merged**